app/ocr.py

```diff
@@ -457,11 +457,60 @@ def get_matches(input_list, topic, full_text):
+        The process starts by adding rules/phrases to SpaCy's Matcher, they were obtained by manually 
+        parsing through case files and finding all sentences related to credibility. 
+        There are three separate rules, narrow, medium and wide, which decrease in the phrasing
+        specificity, this allows for some wiggle room as opposed to searching for exact matches. 
+        All instances of a match are returned by Matcher, so checking whether these objects are empty 
+        or not dictates the output of this function.
         """
-        return "Unknown"
+        # Speciifying phrase patterns / rules to use in SpaCy's Matcher
+        narrow_scope = [[{"LOWER": "court"}, {"LOWER": "finds"},
+                         {"LOWER": "respondent"}, {"LOWER": "generally"},
+                         {"LOWER": "credible"}],
+                        [{"LOWER": "court"}, {"LOWER": "finds"},
+                         {"LOWER": "respondent"}, {"LOWER": "testimony"},
+                         {"LOWER": "credible"}],
+                        [{"LOWER": "court"}, {"LOWER": "finds"}, 
+                         {"LOWER": "respondent"}, {"LOWER": "credible"}]]
+
+        medium_scope = [[{"LOWER": "credible"}, {"LOWER": "witness"}],
+                        [{"LOWER": "generally"}, {"LOWER": "consistent"}],
+                        [{"LOWER": "internally"}, {"LOWER": "consistent"}],
+                        [{"LOWER": "sufficiently"}, {"LOWER": "consistent"}],
+                        [{"LOWER": "testified"}, {"LOWER": "credibly"}],
+                        [{"LOWER": "testimony"}, {"LOWER": "credible"}],
+                        [{"LOWER": "testimony"}, {"LOWER": "consistent"}]]
+
+        wide_scope = [{"LEMMA": {"IN": ["coherent", 
+                                        "possible", 
+                                        "credible", 
+                                        "consistent"]}}]
+        
+        # instantiating Matcher
+        matcher = Matcher(nlp.vocab)
+        
+        # adding each rule to Matcher, then using global function similar() to find 
+        # and store matches in similar_****** variables
+        matcher.add('narrow_cred', narrow_scope)
+        similar_narrow = similar(target_phrases=narrow_scope, file=self.doc)
+
+        matcher.add('medium_cred', medium_scope)
+        similar_medium = similar(target_phrases=medium_scope, file=self.doc)
+
+        matcher.add('wide_cred', wide_scope)
+        similar_wide = similar(target_phrases=wide_scope, file=self.doc)
+        
+        # output logic checks wheteher similar_***** variables are empty or not
+        if similar_narrow:
+            return 'Respondent was found credible'
+
+        elif similar_medium and similar_wide:
+            return 'Respondent was found credible'
+
+        else:
+            return 'Respondent was not found credible'
 
     def check_for_one_year(self) -> bool:

```

Comment:
Can this be deleted?

Reply:
Which part Andrew?

Reply:
Changed it to True and False. Is that better or..?